src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -107,12 +208,52 @@ public List<AssignmentStudentDTO> getStudentAssignments(
             @RequestParam("year") int year,
             @RequestParam("semester") String semester) {
 
-        // TODO remove the following line when done
-
         // return a list of assignments and (if they exist) the assignment grade
-        //  for all sections that the student is enrolled for the given year and semester
-		//  hint: use the assignment repository method findByStudentIdAndYearAndSemesterOrderByDueDate
+        // for all sections that the student is enrolled for the given year and semester
+        // hint: use the assignment repository method
+        // findByStudentIdAndYearAndSemesterOrderByDueDate
+
+        List<Assignment> assignments = assignmentRepository.findByStudentIdAndYearAndSemesterOrderByDueDate(studentId, year, semester);
+        if (assignments == null) {
```

Comment:
Will this ever be null? Did you mean:
```java
if (assignments.isEmpty()) {
```
Though, an empty list probably isn't an error state, so the check might not be needed at all. Perhaps the existence of the enrollment should be checked separately.

Reply:
Suggested change made.